projects/batfish-client/src/org/batfish/client/Client.java

```diff
@@ -315,6 +307,290 @@ private boolean answer(String questionTemplateName, String paramsLine,
       return execute(wItemAs, outWriter);
    }
 
+   /**
+    * For each key in {@code parameters}, validate that its value satisfies the
+    * requirements specified by {@code variables} for that specific
+    * key. Set value to {@code variables} if validation passed.
+    *
+    * @throws BatfishException if the key in parameters does not exist in
+    * variable, or the values in {@code parameters} do not match the
+    * requirements in {@code variables} for that specific key.
+    */
+   static void validateAndSet(Map<String, String> parameters,
+                        Map<String, Variable> variables)
+                        throws BatfishException {
+      BatfishObjectMapper mapper = new BatfishObjectMapper();
+      for (Entry<String, String> e : parameters.entrySet()) {
+         String parameterName = e.getKey();
+         String parameterValue = e.getValue();
+         Variable variable = variables.get(parameterName);
+         if (variable == null) {
+            throw new BatfishException("No variable named: '" + parameterName
+                  + "' in supplied question template");
+         }
+         JsonNode value;
+         try {
+            value = mapper.readTree(parameterValue);
+         }
+         catch (IOException e1) {
+            throw new BatfishException("Variable value is not valid JSON",
+                  e1);
+         }
+         if (value == null) {
+            throw new IllegalStateException(
+                  String.format("Failed to extract JsonNode " +
+                              "for parameter: %s, with variable value: %s.",
+                              parameterName, parameterValue));
+         }
```

Comment:
extract lines 331–344 into a helper and test it. Ensure that the value (that is not valid JSON) is printed when throwing an error :)